src/repository-status-handler-helper.coffee

```diff
@@ -0,0 +1,30 @@
+Task = require './task'
+
+handlerInstance = null
+
+startHandler = ->
+  if not handlerInstance?
+    handlerInstance = new Task require.resolve('./repository-status-handler')
+    terminatedSub = handlerInstance.on "exit", ->
+      terminatedSub.dispose()
+      handlerInstance = null
+    handlerInstance.start()
+
+terminateHandler = ->
+  if handlerInstance?
+    handlerInstance.terminate()
+    handlerInstance = null
+
+refreshStatus = (repoPath, paths) ->
+  startHandler()
+  new Promise (resolve) ->
+    responseSub = handlerInstance.on repoPath, (result) ->
+      responseSub.dispose()
+      resolve(result)
+
+    handlerInstance.send {repoPath, paths}
+
+module.exports = {
+  terminateHandler,
+  refreshStatus
+}
```

Comment:
It would be nice from a style perspective to export a `RepositoryStatusHandler` *class* here, rather than these top-level functions that use a global variable. Then we would instantiate the class in the constructor of `GitRepositoryProvider` (which only gets called once per Atom window) and pass the `RepositoryStatusHandler` instance to the constructor of `GitRepository`.

That said, there are other global variables in Atom, so it's not that big a deal.

Reply:
The intent was to have the handlerInstance variable hidden in this module in such a way where there can be only one per runtime (if my understanding of modules is correct, no matter how many times you require a module, it will only be initialized once per runtime).

The same could be achieved using a class with either static variable, or a class using a module global. Which would you prefer?